recipes-bsp/u-boot/files/0027-arm-dts-add-the-m.2-support.patch

```diff
@@ -0,0 +1,145 @@
+From 09cc8b834d2015915d7176a73444d72db20a9852 Mon Sep 17 00:00:00 2001
+From: chao zeng <chao.zeng@siemens.com>
+Date: Fri, 7 Jan 2022 10:18:07 +0800
+Subject: [PATCH 27/28] arm:dts:add the m.2 support
+
+add the support for the m.2 board based on
+the iot2050 advanced board
+
+Signed-off-by: chao zeng <chao.zeng@siemens.com>
+---
+ arch/arm/dts/Makefile                         |  4 +-
+ .../arm/dts/k3-am6548-iot2050-advanced-m2.dts | 92 +++++++++++++++++++
+ configs/iot2050_pg2_defconfig                 |  2 +-
+ 3 files changed, 96 insertions(+), 2 deletions(-)
+ create mode 100644 arch/arm/dts/k3-am6548-iot2050-advanced-m2.dts
+
+diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
+index b3e2a9c9d7..9ea395dd91 100644
+--- a/arch/arm/dts/Makefile
++++ b/arch/arm/dts/Makefile
+@@ -1119,7 +1119,9 @@ dtb-$(CONFIG_SOC_K3_AM6) += \
+ 	k3-am6528-iot2050-basic.dtb \
+ 	k3-am6528-iot2050-basic-pg2.dtb \
+ 	k3-am6548-iot2050-advanced.dtb \
+-	k3-am6548-iot2050-advanced-pg2.dtb
++	k3-am6548-iot2050-advanced-pg2.dtb \
++	k3-am6548-iot2050-advanced-m2.dtb
++
+ dtb-$(CONFIG_SOC_K3_J721E) += k3-j721e-common-proc-board.dtb \
+ 			      k3-j721e-r5-common-proc-board.dtb \
+ 			      k3-j7200-common-proc-board.dtb \
+diff --git a/arch/arm/dts/k3-am6548-iot2050-advanced-m2.dts b/arch/arm/dts/k3-am6548-iot2050-advanced-m2.dts
+new file mode 100644
+index 0000000000..741b9177da
+--- /dev/null
++++ b/arch/arm/dts/k3-am6548-iot2050-advanced-m2.dts
+@@ -0,0 +1,92 @@
++// SPDX-License-Identifier: GPL-2.0
++/*
++ * Copyright (c) Siemens AG, 2018-2022
++ *
++ * Authors:
++ *   Chao Zeng <chao.zeng@siemens.com>
++ *
++ * AM6548-based (quad-core) IOT2050 M2 variant, Based on Advanced Product Generation 2
++ * 2 GB RAM, 16 GB eMMC, USB-serial converter on connector X30
++ *
++ * Product homepage:
++ * https://new.siemens.com/global/en/products/automation/pc-based/iot-gateways/simatic-iot2050.html
++ */
++
++/dts-v1/;
++
++#include "k3-am6548-iot2050-advanced-common.dtsi"
++#include "k3-am65-iot2050-common-pg2.dtsi"
++
++/ {
++	compatible = "siemens,iot2050-advanced-m2", "ti,am654";
```

Comment:
This compatible does not exist, according to the kernel's dtbs. Also keep in mind that U-Boot should be able to provide an own DTB as input for the UEFI payload in order to be SystemReady.